### PR TITLE
Magento 2: Only set maintenance mode once if needed

### PR DESCRIPTION
**Pull request by schmengler**
Also:
- by adding `magento:maintenance:enable-if-needed` to the task list instead of invoking it directly, it is now possible to use `after()` to add custom tasks to be performed in maintenance mode
- the `maintenance_mode_status_active` check was removed because it is unnecessary and can return wrong results because of caching

- [ ] Bug fix #…?
- [x] New feature?
- [x] BC breaks?
- [ ] Tests added?
- [x] Docs added?

      Please, regenerate docs by running next command:
      $ php bin/docgen

### CODE REVIEW

File: recipe/magento2.php [hunk 1]
```diff
@@ -132,65 +154,28 @@
...
-            invoke('magento:maintenance:enable');
-        }
-
-        run("{{bin/php}} {{release_or_current_path}}/bin/magento setup:upgrade --keep-generated --no-interaction");
-
-        if (get('enable_zerodowntime') && !get('maintenance_mode_status_active')) {
-            invoke('magento:maintenance:disable');
-        }
+    if (get('database_upgrade_needed')) {
+        run("{{bin/php}} {{release_or_current_path}}/{{bin/magento}} setup:upgrade --keep-generated --no-interaction");
```
Comment (peterjaap) [inline recipe/magento2.php:176]: Again, see https://github.com/deployphp/deployer/pull/3460#discussion_r1091092853

Comment (schmengler) [reply recipe/magento2.php:176]: I would like to wait until https://github.com/deployphp/deployer/pull/3460 is merged to fix this

**Review by tdgroot — APPROVED** [review]
LGTM! While it is a minor improvement feature-wise, I think the readability of the recipe has improved a lot. :+1:

File: recipe/magento2.php (hunk 1 above)
Comment (schmengler) [reply recipe/magento2.php]: Rebased and fixed `bin/magento` calls :heavy_check_mark:

**Comment by antonmedv** [comment]
@peterjaap PTAL

**Comment by peterjaap** [comment]
Perfect!

**Review by peterjaap — APPROVED** [review]

**Comment by peterjaap** [comment]
@schmengler could you resolve the magento2.md conflicts? Then it's good to go.

**Comment by schmengler** [comment]
Done @peterjaap @antonmedv

**Comment by peterjaap** [comment]
@antonmedv merge at will

**Pull request merged**